### PR TITLE
[EZP-29121] Added UI Guidelines to documentation, basic structure

### PR DESCRIPTION
**Pull request by inakijv**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [EZP-29121](https://jira.ez.no/browse/EZP-29121)
| Versions      | 2.2

Most important aspects:
Added UI Guidelines to documentation, basic structure:
- Design Principles section: Design principles, Philosophy and Accessibility;
- Resources section: Typography, Colors and Icons;
- Resources section: buttons component (basic buttons).

design principles - ez platform developer documentation
typography - ez platform developer documentation
colors - ez platform developer documentation
icons - ez platform developer documentation 1
buttons - ez platform developer documentation 1

### CODE REVIEW

**Comment by inakijv** [comment]
Ping @DominikaK

**Comment by damianz5** [comment]
`mdx_ez_code_example` should be moved to github ezsystems namespace, besides that looks great!
ping @SylvainGuittard what do you think?

File: docs/guidelines/design_principles/Philosophy.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+The design philosophy underpins everything we do as a Product and Engineering Teams. It informs the way our products look, the way they behave and the way we work as a team.
```
Comment (DominikaK) [inline docs/guidelines/design_principles/Philosophy.md]: And missing heading again: `# Philosophy`

**Comment by inakijv** [comment]
@DominikaK updated based on comments. Thanks!

**Comment by SylvainGuittard** [comment]
> mdx_ez_code_example should be moved to github ezsystems namespace, besides that looks great!
@damianz5 I totally agree.

**Comment by DominikaK** [comment]
@inakijv, I adapted the text to fit better with the style of the rest of the dev doc. Please take a look to make sure that didn't twist your meaning.

Also, with the styling of headings there's a small problem with how markdown renders the links:
guidelines_brken_headings

That's a small thing, but maybe @damianz5 has an idea how to fix it?

**Comment by inakijv** [comment]
@DominikaK thank you. Updated based on discussed last week. I need a final clear on this :-) and ask engineers for their feedback. Thanks!

**Comment by sunpietro** [comment]
Why is it so big? I'd prefer making smaller changes in a multiple PRs

**Comment by sunpietro** [comment]
Another question: do you use Bootstrap 3 and Bootstrap 4 on the same website? It's going to be a maintainance hell. As far as I know, our examples need Bootstrap 4, but the guideline is built with Bootstrap 3, right? So why didn't you display our code samples in iframes? It would separate the environments perfectly. No need to prefix Bootstrap, no naming conventions collisions.

**Comment by inakijv** [comment]
@sunpietro answering to your questions:

- MKDOCs still uses Bootstrap 3, and we have had to find a solution combining both. Among all options discussed, this one was the most suitable for the needs we had;
- Regarding the length of the PR, it sets the basic structure for this section (Components, Resources, Design principles) within Documentation site. I am more than happy to discuss any other aspects regarding the structure or the content (actually I believe it would be terrific);
- The PRs will be smaller from now on due to they are going to be focused on each of the basic UI components, your input guys is really valuable for the `Buttons` section, for instance, due to the developer perspective that needs to be included here.

We are building a set of UI components, that is also a three-way collaboration (doc, engineering and pm teams) that needs to include all these views.

**Comment by sunpietro** [comment]
@inakijv what about using iframes to display demo samples?

**Comment by damianz5** [comment]
`mkdocs-material` is not using bootstrap in any version (https://squidfunk.github.io/mkdocs-material/),
iframes are not supported by mkdocs (https://github.com/mkdocs/mkdocs/issues/1246), mkdocs will not generate partials, only the pages listed in mkdocs.yml. 
Search engine will not be able to find and index content inside iframes - we will not be able to search

**Comment by sunpietro** [comment]
So this is a dirty, but the only working way. I see.

**Pull request merged**